### ISSUE TITLE
[Regression] Crash on exit/switching m3u urls after 1.9.6.0

### ISSUE THREAD

**Issue by The-Solutor** [clsid2/mpc-hc#2337]
Hi.

Something I have observed since the release of 1.9.7 but I haven't reported hoping that was a temporary problem that would be fixed on following releases, but as 2.1.2 the problem is still there.

In short starting from 1.9.7.0 often MPC crashes when switching from a playlist url (in my case are often channels to my TV server) to another, or when closing the program.

I'm facing it on a old AMD machine where I mostly use the overlay mixer or the VMR7 output. Given the machine doesn't handle properly the DXVA2 acceleration, so the overlay mixer greatly reduces the CPU utilization.

Maybe a bit of corner case, but still anything older than 1.9.7 works perfectly.

P.S. the relevant dump is here

https://drdump.com/UploadedReport.aspx?DumpID=124066148

**Comment by clsid2** [comment]
NEVER wait with reporting problems!

Might be due to https://github.com/clsid2/mpc-hc/commit/c5e32a2766715e88f47e1bdd2b2943673e1e4fcb

Are you are to trigger the problem with other renderers?

**Comment by The-Solutor** [comment]
Thanks for the quick reply.

About waiting for report, I have expressed myself poorly...

I mean I noticed the problem while ago, not at the time of 1.9.7 release.

But when I noticed the problem I was unsure it was related to something wrong on my system or something.

Today I had a bit of free time and I tested multiple OSes and also MPC-HC version by version trying to narrow down the problem until I was sure when the problem appeared for the first time.

About the renderer, yes it seem like the problem is limited to overlay mixer and vmr7, for sure EVR works (other renderers have problems unrelated to MPC-HC, that trigger the restart of the video driver)

P.S. Apropos of overlay mixer, not sure if that's the intended behavior, but why I cant select the overlay mixer in x64 version?

If I select it in the x86 version then it works w/o any problem in the x64 flavour, but in the settings I can't see it listed (in 2.1.2) or I see it listed as "unavailable" in older versions.

**Comment by clsid2** [comment]
There is no 64-bit version of overlay mixer.

**Comment by The-Solutor** [comment]
That's what I guessed

Then the question is why it works anyway?

IIRC overlay mixer in x64 was used to be selectable on the original MPC-HC 1.7.xx and also in very old builds of current branch

(it's clearly working, I see it indirectly in the way the menu overlaps the video windows, the CPU usage and alike)

**Comment by clsid2** [comment]
It falls back to using system default (vmr7).

**Comment by The-Solutor** [comment]
Thanks.

That's the explanation that I gave myself before asking.

But still definitely the behavior of overlay mixer over VMR7 is different (on both x86 and x64).

If I select overlay mixer everything works as expected. If I select VMR7 the video driver restarts once, then everything works.

In this very machine, in both modes I have black screenshots, in both modes I have low CPU usage, in both modes I have a black border around something that overlaps the video frame, so I have no other way to distinguish them.

In short, aside the academical curiosity, the point is that to get the best from the x64 version I have to use the option window from the x86 one.

**Comment by clsid2** [comment]
Menu > Play > Filters
The Pin Info tab of the renderer filter shows what it is using.

Default VMR7 is {B87BEB7B-8D29-423F-AE4D-6582C10175AC}.
Another one is {70E102B0-5556-11CE-97C0-00AA0055595A}. You can use that one by setting "video renderer" (first one of two) as a preferred external filter.

**Comment by clsid2** [comment]
Try 2.1.2.18 development build.

**Comment by The-Solutor** [comment]
Yeeeppp!!!

Seem fixed. 

Thank you very much.
